api/utils.py

```diff
@@ -14,6 +14,7 @@
     HttpResponse,
     QueryDict,
 )
+from django.shortcuts import _get_queryset
```

Comment:
The `_` in the name suggests that this is a private function. Do you feel like this is stable enough that we won’t get rug-pulled in future versions?

Is there a public function that we could use instead?

Reply:
I copied exact contents of `get_object_or_404` function and it was using this method to get the queryset. I only changed the raised response upon failure. Since this acts like `get_object_or_404`, I think it is okay to use an internal function of Django in this case.

I also checked up to date version and older versions of this function and I think this function stability level is 5/5.

https://github.com/django/django/blob/d22ba0763086f71fb6df8502162ea63944d166b1/django/shortcuts.py#L51